shell/config/uiplugins.js

```diff
@@ -3,44 +3,37 @@ import semver from 'semver';
 // Version of the plugin API supported
 // here we inject the current shell version that we read in vue.config
 export const UI_EXTENSIONS_API_VERSION = process.env.UI_EXTENSIONS_API_VERSION;
+const MINIMUM_MANDATORY_EXTENSIONS_API_VERSION = `> ${ UI_EXTENSIONS_API_VERSION }`;
+
 export const UI_PLUGIN_HOST_APP = 'rancher-manager';
 
 export const UI_PLUGIN_BASE_URL = '/v1/uiplugins';
-
 export const UI_PLUGIN_NAMESPACE = 'cattle-ui-plugin-system';
 
 // Annotation name and value that indicate a chart is a UI plugin
-export const UI_PLUGIN_ANNOTATION_NAME = 'catalog.cattle.io/ui-component';
-export const UI_PLUGIN_ANNOTATION_VALUE = 'plugins';
-
-export const UI_PLUGIN_OPERATOR_CRD_CHART_NAME = 'ui-plugin-operator-crd';
-export const UI_PLUGIN_OPERATOR_CHART_NAME = 'ui-plugin-operator';
-
-export const UI_PLUGIN_CHARTS = [
-  UI_PLUGIN_OPERATOR_CHART_NAME,
-  UI_PLUGIN_OPERATOR_CRD_CHART_NAME,
-];
-
-// Expected chart repo name for the UI Plugins operator
-export const UI_PLUGIN_OPERATOR_REPO_NAME = 'rancher-charts';
-
-// Info for the Helm Chart Repository that we will add
-export const UI_PLUGINS_REPO_NAME = 'rancher-ui-plugins';
-
-export const UI_PLUGINS_REPO_URL = 'https://github.com/rancher/ui-plugin-charts';
-export const UI_PLUGINS_REPO_BRANCH = 'main';
-
-// Info for the Helm Chart Repo for Partner Extensions
-export const UI_PLUGINS_PARTNERS_REPO_NAME = 'partner-extensions';
-
-export const UI_PLUGINS_PARTNERS_REPO_URL = 'https://github.com/rancher/partner-extensions';
-export const UI_PLUGINS_PARTNERS_REPO_BRANCH = 'main';
-
-// Info for the Helm Chart Repo for Community Extensions
-export const UI_PLUGINS_COMMUNITY_REPO_NAME = 'community-extensions';
+export const UI_PLUGIN_ANNOTATION = {
+  NAME:  'catalog.cattle.io/ui-component',
+  VALUE: 'plugins',
+};
 
-export const UI_PLUGINS_COMMUNITY_REPO_URL = 'https://github.com/rancher/community-extensions';
-export const UI_PLUGINS_COMMUNITY_REPO_BRANCH = 'main';
+// Info for the Helm Chart Repositories
+export const UI_PLUGINS_REPOS = {
+  OFFICIAL: {
+    NAME:   'rancher-ui-plugins',
+    URL:    'https://github.com/rancher/ui-plugin-charts',
+    BRANCH: 'main',
```

Comment:
Question only. in `master` these will point to `main`. In future versions these will change to one related to rancher version branches of those repos?

Reply:
Tbh, I don't see it evert changing in the future. 

These are consumed by our end users and if we couldn't change the `ui-plugin-examples` repo from `main` to `gh-pages`, these won't be allowed for sure.